pkg/endpoints/dashboard.go

```diff
@@ -130,78 +106,177 @@ func IsOpenShift(r Resource, installedNamespace string) bool {
-			if potentialVersion != "" {
-				version = potentialVersion
+	if version == "" && thingSearchingFor == "pipelines" {
+		for _, deployment := range deployments.Items {
+			deploymentLabels := deployment.GetLabels()
+			labelsToCheck := []string{"pipeline.tekton.dev/release", "version"} // To handle both beta and pre-beta versions
+			for _, label := range labelsToCheck {
+				potentialVersion := deploymentLabels[label]
+				if potentialVersion != "" {
+					version = potentialVersion
+				}
 			}
-		}
 
-		// For Tekton Pipelines 0.9.0 - 0.9.2
-		if version == "" {
-			deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
-			if strings.Contains(deploymentImage, "pipeline/cmd/controller") && strings.Contains(deploymentImage, ":") && strings.Contains(deploymentImage, "@") {
-				s := strings.SplitAfter(deploymentImage, ":")
-				if strings.Contains(s[1], "@") {
-					t := strings.Split(s[1], "@")
-					version = t[0]
+			if namespace == "openshift-pipelines" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "openshift-pipeline/tektoncd-pipeline-controller") && strings.Contains(deploymentImage, ":") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if s[1] != "" {
+						version = s[1]
+					}
+				}
+			}
+
+			deploymentAnnotations := deployment.Spec.Template.GetAnnotations()
+			annotationsToCheck := []string{"pipeline.tekton.dev/release", "tekton.dev/release"} // To handle 0.10.0 and 0.10.1
+			for _, label := range annotationsToCheck {
+				potentialVersion := deploymentAnnotations[label]
+				if potentialVersion != "" {
+					version = potentialVersion
+				}
+			}
+
+			// For Tekton Pipelines 0.9.0 - 0.9.2
+			if version == "" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "pipeline/cmd/controller") && strings.Contains(deploymentImage, ":") && strings.Contains(deploymentImage, "@") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if strings.Contains(s[1], "@") {
+						t := strings.Split(s[1], "@")
+						version = t[0]
+					}
 				}
 			}
 		}
 	}
 
 	return version
 }
+
+// Get triggers version
+func GetTriggersVersion(r Resource, isOpenShift bool) string {
+	version := ""
+	namespacesToCheck := []string{"tekton-pipelines"}
+
+	if isOpenShift {
+		namespacesToCheck = append(namespacesToCheck, "openshift-pipelines")
+	}
+
+	// Go through possible namespaces Tekton Triggers is installed in
+	// For each namespace if the version is empty then it goes through to get the deployments within that namespace to try and locate the Tekton Triggers deployment to get the version
+	for _, versionAttempt := range namespacesToCheck {
+		if version == "" {
+			version = getDeployments(r, "triggers", versionAttempt)
+		}
+	}
+
+	if version == "" {
+		logging.Log.Error("Error getting the Tekton Triggers deployment version. Version is unknown")
+		version = "UNKNOWN"
```

Comment:
This is the only change I made to @CarolynMabbott's original back end code from https://github.com/tektoncd/dashboard/commit/fa128ccaa4285ad25f270a6585abb1473b632827